src/jit/codegenxarch.cpp

```diff
@@ -1958,6 +1956,7 @@ CodeGen::genCodeForTreeNode(GenTreePtr treeNode)
 #endif // !defined(_TARGET_64BIT_)
     case GT_ADD:
     case GT_SUB:
+        genConsumeOperands(treeNode->AsOp());
         genCodeForBinary(treeNode);
```

Comment:
Suggestion: We can move out even genProduceReg() call also out of genCodeForBinary().  That way both genConsume/ProduceReg() calls will be the caller's responsibility. 

Will leave it up to you.